File: demos/ruby/data_structures_demo.rb
require "set"

Node = Data.define(:value, :children)

values = ["parse", "validate", "render"]
scores = { "Ada" => 95, "Lin" => 91 }
tags = Set.new(%w[docs code docs])
queue = values.dup
tree = Node.new("root", [Node.new("left", []), Node.new("right", [])])

puts "array=#{values.join(',')}"
puts "queue-first=#{queue.shift}"
puts "scores=#{scores.map { |key, value| "#{key}:#{value}" }.join(',')}"
puts "tags=#{tags.to_a.sort.join(',')}"
puts "tree-children=#{tree.children.length}"

File: demos/python/data_structures_demo.py
from collections import deque
from dataclasses import dataclass
import heapq


@dataclass(frozen=True)
class Node:
    value: str
    children: tuple["Node", ...] = ()


items = ["parse", "validate", "render"]
queue = deque(items)
scores = {"Ada": 95, "Lin": 91}
tags = {"docs", "code", "docs"}
tasks = [(2, "docs"), (1, "tests")]
heapq.heapify(tasks)
tree = Node("root", (Node("left"), Node("right")))

print(f"list={items}")
print(f"queue-first={queue.popleft()}")
print(f"scores={list(scores.items())}")
print(f"tags={sorted(tags)}")
print(f"next={heapq.heappop(tasks)[1]}")
print(f"tree-children={len(tree.children)}")
